rust/src/dns/dns.rs

```diff
@@ -773,14 +776,24 @@ pub unsafe extern "C" fn rs_dns_parse_response_tcp(flow: *const core::Flow,
 }
 
 #[no_mangle]
-pub extern "C" fn rs_dns_tx_get_alstate_progress(_tx: *mut std::os::raw::c_void,
-                                                 _direction: u8)
+pub unsafe extern "C" fn rs_dns_tx_get_alstate_progress(_tx: *mut std::os::raw::c_void,
+                                                 direction: u8)
                                                  -> std::os::raw::c_int
 {
     // This is a stateless parser, just the existence of a transaction
-    // means its complete.
+    // means its complete. However this is only true for UDP and not TCP
+    let tx = cast_pointer!(_tx, DNSTransaction);
     SCLogDebug!("rs_dns_tx_get_alstate_progress");
-    return 1;
+    if tx.is_tcp {
+        if direction & core::STREAM_TOSERVER != 0 && tx.request.is_some() {
+            return 1;
+        } else if direction & core::STREAM_TOCLIENT != 0 && tx.response.is_some() {
+            return 1;
+        }
```

Comment:
Hello, very soon clippy is going to complain about this pattern https://rust-lang.github.io/rust-clippy/master/index.html#if_same_then_else
Can we combine these statements into one so that we only return 1 once?